### PR TITLE
HPO: DLC mode support nas&oss at same time

### PR DESCRIPTION
**Pull request by weidankong**
### Description ###
HPO DLC mode update.
was: able to attach NAS data source only.
now: can also attach an OSS data source at the same time.

### Checklist ###
  - [x] test case: verified with Alibaba PAI DLC
  - [x] doc: updated.

### How to test ###
1. create an NAS data source id on [dlc](https://pai-dlc.console.aliyun.com/)
2. create an OSS data source id on dlc as well, this is optional.
3. create your config.yml, ref: examples/trials/mnist-pytorch/config_dlc.yml 
4. create your job.

### CODE REVIEW

**Review by ultmaster — APPROVED** [review]

File: ts/nni_manager/config/dlc/dlcUtil.py [hunk 1]
```diff
@@ -41,6 +43,13 @@
         data_source_id=args.nas_data_source_id,
     )
 
+    oss = None
+    if (args.oss_data_source_id):
```
Comment (cruiseliu) [inline ts/nni_manager/config/dlc/dlcUtil.py:47]: No parentheses please.

File: ts/nni_manager/config/dlc/dlcUtil.py [hunk 2]
```diff
@@ -41,6 +43,13 @@
...
+        oss = DataSourceItem(
+            data_source_type = 'oss',
+            data_source_id=args.oss_data_source_id,
```
Comment (cruiseliu) [inline ts/nni_manager/config/dlc/dlcUtil.py:50]: Please make spaces around `=` consistent in these two lines.

File: ts/nni_manager/training_service/reusable/dlc/dlcClient.ts [hunk 3]
```diff
@@ -46,11 +51,16 @@ export class DlcClient {
         this.image = image;
         this.region = region;
         this.nasDataSourceId = nasDataSourceId;
+        if (ossDataSourceId !== undefined)
```
Comment (cruiseliu) [inline ts/nni_manager/training_service/reusable/dlc/dlcClient.ts:54]: Please add braces.

File: ts/nni_manager/config/dlc/dlcUtil.py (hunk 2 above)
Comment (weidankong) [reply ts/nni_manager/config/dlc/dlcUtil.py:50]: accept, changed.

File: ts/nni_manager/config/dlc/dlcUtil.py (hunk 1 above)
Comment (weidankong) [reply ts/nni_manager/config/dlc/dlcUtil.py]: sure, updated.

File: ts/nni_manager/training_service/reusable/dlc/dlcClient.ts (hunk 3 above)
Comment (weidankong) [reply ts/nni_manager/training_service/reusable/dlc/dlcClient.ts]: added, thanks.

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**